### PR TITLE
processor/sourcemanager(ticdc): add sorter stats

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

See: https://github.com/pingcap/tiflow/issues/5928#issuecomment-1318253453

### What is changed and how it works?
add sorter stats.

#### Tests 

 - Manual test (add detailed scripts or steps below)

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sourcemanager/engine/pebble/event_sorter.go [hunk 1]
```diff
@@ -298,8 +357,12 @@ type eventWithTableID struct {
 
 type tableState struct {
 	ch              *chann.Chann[eventWithTableID]
-	sortedResolved  uint64 // indicates events are ready for fetching.
-	pendingResolved uint64 // events are resolved but not sorted.
+	sortedResolved  atomic.Uint64 // indicates events are ready for fetching.
+	pendingResolved atomic.Uint64 // events are resolved but not sorted.
+	// For statistics.
+	maxReceivedCommitTs   atomic.Uint64
+	maxReceivedResolvedTs atomic.Uint64
```
Comment (hicqu) [inline cdc/processor/sourcemanager/engine/pebble/event_sorter.go:364]: Seems it's same with `pendingResolved`, right?

**Comment by hicqu** [comment]
Rest LGTM.

**Review by hicqu — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

File: cdc/processor/sourcemanager/engine/memory/event_sorter.go [hunk 2]
```diff
@@ -131,13 +131,25 @@ func (s *EventSorter) CleanByTable(tableID model.TableID, upperBound engine.Posi
...
-// Close implements sorter.EventSortEngine.
+// GetStatsByTable implements engine.SortEngine.
+func (s *EventSorter) GetStatsByTable(tableID model.TableID) engine.TableStats {
+	log.Panic("GetStatsByTable should never be called")
+	return engine.TableStats{}
+}
+
+// RemainEvents implements engine.SortEngine.
+func (s *EventSorter) RemainEvents() int64 {
+	log.Panic("RemainEvents should never be called")
```
Comment (hicqu) [inline cdc/processor/sourcemanager/engine/memory/event_sorter.go:148]: Why still panic?

Comment (hicqu) [reply cdc/processor/sourcemanager/engine/memory/event_sorter.go:148]: I see. It's just used in tests.

File: cdc/processor/sourcemanager/engine/pebble/event_sorter.go [hunk 3]
```diff
@@ -280,6 +335,9 @@ func (s *EventIter) Next() (event *model.PolymorphicEvent, pos engine.Position,
 		}
 		event, s.headItem = s.headItem, event
 	}
+	if event != nil {
+		s.state.remainEvents.Add(-1)
```
Comment (hicqu) [inline cdc/processor/sourcemanager/engine/pebble/event_sorter.go:339]: If redo is enabled, an event can be pushed 1 time but fetched 2 times.
So I suggest to record input events in `SortEngine`, and record fetched events in `tableSinkWrapper`.

**Review by nongfushanquan — APPROVED** [review]

File: cdc/processor/sinkmanager/table_sink_wrapper.go [hunk 4]
```diff
@@ -89,6 +94,12 @@ func (t *tableSinkWrapper) updateReceivedSorterResolvedTs(ts model.Ts) {
 	t.receivedSorterResolvedTs.Store(ts)
 }
 
+func (t *tableSinkWrapper) updateReceivedSorterCommitTs(ts model.Ts) {
+	if ts > t.receivedSorterCommitTs.Load() {
```
Comment (hicqu) [inline cdc/processor/sinkmanager/table_sink_wrapper.go:98]: Seems if redo is enabled, both redo_log_worker and table_sink_worker will call this. I think if redo is enabled, only redo_log_worker needs to call this.

**Review by hicqu — APPROVED** [review]

**Pull request merged**